core/CronArchive/SegmentArchivingRequestUrlProvider.php

```diff
@@ -138,6 +158,15 @@ private function getCreatedTimeOfSegment($idSite, $segmentDefinition)
                 if ($createdTime->getTimestamp() < $earliestCreatedTime->getTimestamp()) {
                     $earliestCreatedTime = $createdTime;
                 }
+
+                if (!empty($segment['ts_last_edit'])) {
+                    $lastEditTime = Date::factory($segment['ts_last_edit']);
+                    if ($latestEditTime === null
+                        || $latestEditTime->getTimestamp() < $lastEditTime->getTimestamp()
```

Comment:
Do I get this right there could be multiple segments matching the segment definition? Just asking for my personnel interest :)